### PR TITLE
fix: set context data more safely

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/assets_context.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ class Assets {
   getContext(data) {
     data = safeStringify(data || this.assetsContext);
     let ret = `<div id="${CONTEXT_TEMPLATE_ID}" style="display:none">${data}</div>\n`;
-    ret += `<script>window.${this.config.contextKey} = JSON.parse(document.getElementById('${CONTEXT_TEMPLATE_ID}').textContent || '{}');</script>`;
+    ret += `<script>window.${this.config.contextKey} = JSON.parse(decodeURIComponent(window.atob(document.getElementById('${CONTEXT_TEMPLATE_ID}').textContent)) || '{}');</script>`;
```
[the remark this answers is not in the document]
Comment (dead-horse) [reply lib/assets_context.js]: 不编码还是会有问题，语雀遇到了

[the remark this answers is not in the document]
Comment (popomore) [reply lib/assets_context.js]: https://github.com/davidchambers/Base64.js/blob/master/base64.js

[the remark this answers is not in the document]
Comment (popomore) [reply lib/assets_context.js]: https://github.com/beatgammit/base64-js/blob/master/index.js

**Comment by popomore** [comment]
我先合并了，后面把 base64 自己实现下

**Pull request merged**